### PR TITLE
[GPU] Raja Cuda Wrap 2D & 3D using Kernel Policies [raja-wrap]

### PR DESCRIPTION
**Pull request by camierjs**
Uses `RAJA::kernel` to set up the underlying grid.
`RajaCudaWrap2D` uses the `Z` dimension for batches.

Can be tested with the [CEED benchmarks](https://github.com/CEED/benchmarks), with this command:
```./go.sh cuda_arch=sm_70 -p 1 -n 1 -c lassen -m xlc -r tests/mfem4_bps/bp3.sh mfem_branch=raja-wrap min_p=6 max_p=6 max_dofs_proc=15000000 num_proc_build=32 'mfem_devs="raja-cuda"'```
just to see if it is good, then:
```./go.sh cuda_arch=sm_70 -u -p 1 -n 1 -c lassen -m xlc -r tests/mfem4_bps/bp3.sh mfem_branch=raja-wrap min_p=1 max_p=6 max_dofs_proc=15000000 num_proc_build=32 'mfem_devs="cuda raja-cuda"' > tests/mfem4_bps/bp3.raja-wrap``` and ```python postprocess-plot-2.py bp3.raja-wrap```.

### CODE REVIEW

**Comment by tzanio** [comment]
Thanks for letting us know about this @artv3 !

File: general/raja.hpp [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+__device__ __forceinline__ int globalIdx3D() { return blockIdx.x; }
+
+template <bool batch, typename Iterator, typename DBODY, typename IndexType>
+__global__ void RajaKernelND(DBODY body, Iterator I, IndexType N)
+{
+   const auto k = static_cast<IndexType>(batch ? globalIdx2D():globalIdx3D());
+   if (k >= N) { return; }
+   body(I[k]);
+}
```
Comment (artv3) [inline general/raja.hpp]: I have an idea.... try replacing lines 38-40 with 
 ```
const int k = blockIdx.x;
  if (k >= N) { return; }
  body(k); 
```
like you have in your CUDA code and try a 3D problem. Just for benchmarking purposes....

**Comment by camierjs** [comment]
plot2_MFEM_cuda_bp3_lassen_xlc_N001_pn1
plot2_MFEM_raja-cuda_bp3_lassen_xlc_N001_pn1

**Comment by camierjs** [comment]
These are BP3, with the old hand-made `RAJA::cuda::forallND`.

File: general/raja.hpp (hunk 1 above)
Comment (camierjs) [reply general/raja.hpp]: Tried, but it does not really change the performances.

**Pull request merged**